pgmpy/inference/Sampling.py

```diff
@@ -0,0 +1,229 @@
+            None if no evidence
+        size: int
+            size of sample to be generated
+
+        Returns
+        -------
+        sampled: pandas.DataFrame
+            the generated samples
+
+        Examples
+        --------
+        >>> from pgmpy.models.BayesianModel import BayesianModel
+        >>> from pgmpy.factors.CPD import TabularCPD
+        >>> from pgmpy.factors.Factor import State
+        >>> from pgmpy.inference.Sampling import BayesianModelSampling
+        >>> student = BayesianModel([('diff', 'grade'), ('intel', 'grade')])
+        >>> cpd_d = TabularCPD('diff', 2, [[0.6], [0.4]])
+        >>> cpd_i = TabularCPD('intel', 2, [[0.7], [0.3]])
+        >>> cpd_g = TabularCPD('grade', 3, [[0.3, 0.05, 0.9, 0.5], [0.4, 0.25,
+        ...                0.08, 0.3], [0.3, 0.7, 0.02, 0.2]],
+        ...                ['intel', 'diff'], [2, 2])
+        >>> student.add_cpds(cpd_d, cpd_i, cpd_g)
+        >>> inference = BayesianModelSampling(student)
+        >>> evidence = {'diff': State(var='diff', state=0)}
+        >>> inference.rejection_sample(evidence, 2)
+                grade       intel
+        0  (grade, 2)  (intel, 1)
+        1  (grade, 0)  (intel, 0)
+        """
+        def check_if_consistent(dict1, dict2):
+            """
+            Checks if two assignments of variables contradict each other.
+
+            Parameters
+            ----------
+            dict1, dict2: dicts
+                key, value pairs denoting the variable and the assignment to it
+
+            Returns
+            -------
+            boolean: True/False
+            """
+            common_keys = set(dict1.keys()) - set(dict2.keys())
+            for k in common_keys:
+                if dict1[k] != dict2[k]:
+                    return False
+            return True
+
+        if evidence is None:
+            return self.forward_sample(size)
+
+        query_vars = list(set(self.topological_order) - set(evidence.keys()))
+        sampled = DataFrame(index=range(0, size),
+                            columns=query_vars)
+        i = 0
+        while i < size:
+            particle = {}
+            for node in self.topological_order:
+                cpd = self.cpds[node]
+                if cpd.evidence:
+                    evid = []
+                    for var in cpd.evidence:
+                        evid.append(particle[var])
+                    weights = cpd.reduce(evid, inplace=False).values
+                else:
+                    weights = cpd.values
+                particle[node] = sample_discrete(cpd.variables[cpd.variable],
+                                                 weights)
+            if check_if_consistent(evidence, particle):
+                # reject if the sample contradicts the evidence
+                sampled.loc[i] = [particle[node] for node in query_vars]
+                i += 1
+        return sampled
+
+    def likelihood_weighted_sample(self, evidence=None, size=1):
```

Comment:
@pratyakshs I think you should be able to make similar changes as per my comment for `forward_sampling` in this method also.